Added support for customizable tooltip

determinationlove opened the pull request:
## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

<!-- Please check the one that applies to this PR using "x". -->
```
[ ] Bugfix
[X] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
<!-- Please describe the current behavior that you are modifying, or link to a relevant issue. -->

The current tooltip simply adds a `<title>` svg tag to the Label element.
But users usually need tooltips to view richer information than simply displaying labels.
React has a lot of tooltip libs, and we need to be able to support using these libs directly without having to reinvent the wheel ourselves.

Issue Number: #184 

## What is the new behavior?

I changed the `Node` component and added a new prop `tooltip`. `Tooltip` components from other libs can be passed directly into `Node`.

I placed the preview results of this feature in `CustomElements` of `Basic.stories.tsx` and `NodeRearranging` of `Drag.stories.tsx`.
The third-party tooltip lib used for demonstration uses [**Antd**](https://ant.design/).

## Does this PR introduce a breaking change?
```
[ ] Yes
[X] No
```

<!-- If this PR contains a breaking change, please describe the impact and migration path for existing applications below. -->


## Other information

This is my first time contributing to an open source project 🙏

amcdnl commented on package.json:
```diff
@@ -49,6 +49,7 @@
   "homepage": "https://github.com/reaviz/reaflow#readme",
   "dependencies": {
     "@juggle/resize-observer": "^3.4.0",
+    "antd": "^5.22.0",
```
Can you use `reablocks` tooltip? This projecty already uses it.

determinationlove replied:
I tried it but I have a problem when using component of reablocks in reaflow

I'm trying to import reablocks `Popover` into `Basic.stories.tsx`
Error screams at me right now: 
```
useTheme must be used within a ThemeProvider
```
I really don’t know how to solve this Error. using other third-party libraries to achieve custom tooltips doesn’t have this problem at all and can be achieved quickly.

amcdnl replied:
Take a look at how we did it in reaviz - https://github.com/reaviz/reaviz/blob/master/src/common/Tooltip/ChartTooltip.tsx#L41

determinationlove replied:
ok, I've changed `Node`'s tooltip props demo to use `reablocks`.

Added `tailwindcss`
Modify `postcss.config.cjs` to make Tailwind style take effect
Modify `.storybook/preview.tsx` to make `stories` use `reablocks` Theme

determinationlove committed:
Added support for customizable tooltip

Added support for customizable tooltip (with reablocks or else component library)

determinationlove commented:
I saw that reaflow has released a new version (v5.4.1),
I will update the PR to the latest version and force push again 🥲

determinationlove commented:
@amcdnl please review the PR 🙏
